framework/src/meshgenerators/ParsedSubdomainMeshGenerator.C

```diff
@@ -8,45 +8,20 @@
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
 #include "ParsedSubdomainMeshGenerator.h"
-#include "Conversion.h"
-#include "MooseMeshUtils.h"
-#include "CastUniquePointer.h"
-
-#include "libmesh/fparser_ad.hh"
-#include "libmesh/elem.h"
 
 registerMooseObject("MooseApp", ParsedSubdomainMeshGenerator);
 
 InputParameters
 ParsedSubdomainMeshGenerator::validParams()
 {
-  InputParameters params = MeshGenerator::validParams();
-  params += FunctionParserUtils<false>::validParams();
+  InputParameters params = ParsedSubdomainGeneratorBase::validParams();
 
-  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
   params.addRequiredParam<std::string>("combinatorial_geometry",
```

Comment:
I would move this to the base class with a generic "expression" name then just rename it in the derived class as needed

That way the logic for the expression string can move to the base class